src/cupbearer/data/transforms.py

```diff
@@ -1,10 +1,12 @@
 from abc import ABC, abstractmethod
 from dataclasses import dataclass
-from typing import Optional
+from typing import Any, Optional
 
 import torch
 import torchvision.transforms.functional as F
 
+Sample = tuple[torch.Tensor] | tuple[torch.Tensor, Any]
```

Comment:
In practice, this is correct, but it seems inconsistent with the type annotations for `__rest_call__`, should be `tuple[torch.Tensor, Any, ...]` for the second option. At that point, we might as well use `tuple[torch.Tensor, ...]`. Though I think arguably we should have a more consistent convention here anyway, not sure what it should be

Reply:
I had `tuple[torch.Tensor, *tuple[Any, ...]]` before I realized that that gives SyntaxError in Python<3.11. Would be an option if we bump version requirements.

Alternatively, perhaps be restrictive to `tuple[torch.Tensor, ...]` instead would be best.

Reply:
I made the typehint less specific than it should be for __call__ but type checkers should be correct as long as they pick up on sample being split and sent to the separate methods form img and rest.